### PR TITLE
rpc: add `invokecontractverify` RPC-method

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Not exactly the https://github.com/neo-project/neo-modules/pull/378#issuecomment-722652859 but changes ported from https://github.com/neo-project/neo-modules/pull/409.

### CODE REVIEW

**Comment by roman-khimov** [comment]
@fyrchik, please review.

File: pkg/rpc/client/rpc_test.go [hunk 1]
```diff
@@ -806,6 +807,46 @@ var rpcClientTestCases = map[string][]rpcClientTestCase{
 			},
 		},
 	},
+	"invokecontractverify": {
```
Comment (fyrchik) [inline pkg/rpc/client/rpc_test.go:810]: Can we add this test without prebuilt response to `server/client_test.go` instead?

File: pkg/rpc/server/server_test.go [hunk 2]
```diff
@@ -773,6 +773,62 @@ var rpcTestCases = map[string][]rpcTestCase{
...
+				require.True(t, ok)
+				assert.NotNil(t, res.Script)
+				assert.Equal(t, "HALT", res.State)
+				assert.NotEqual(t, 0, res.GasConsumed)
+				assert.Equal(t, true, res.Stack[0].Value().(bool))
+			},
+		},
+		{
+			name:   "unknown contract",
+			params: fmt.Sprintf(`["%s", []]`, strings.ReplaceAll(verifyContractHash, "a", "b")),
```
Comment (fyrchik) [inline pkg/rpc/server/server_test.go:818]: This is a bit unreliable, smth like `util.Uint160{}.String()` is more suitable, I believe.

File: pkg/rpc/server/server.go [hunk 3]
```diff
@@ -1121,12 +1123,52 @@ func (s *Server) invokescript(reqParams request.Params) (interface{}, *response.
...
+		return nil, response.NewInternalServerError("can't create invocation script", err)
+	}
+	tx.Script = script
+	return s.runScriptInVM(script, tx, trigger.Verification)
 }
 
 // runScriptInVM runs given script in a new test VM and returns the invocation
 // result.
-func (s *Server) runScriptInVM(script []byte, tx *transaction.Transaction) (*result.Invoke, *response.Error) {
+func (s *Server) runScriptInVM(script []byte, tx *transaction.Transaction, t trigger.Type) (*result.Invoke, *response.Error) {
```
Comment (fyrchik) [inline pkg/rpc/server/server.go:1171]: Let the trigger be first like in other places?

File: pkg/rpc/server/server.go [hunk 4]
```diff
@@ -1121,12 +1123,52 @@ func (s *Server) invokescript(reqParams request.Params) (interface{}, *response.
...
+	}
+	if len(reqParams) > 2 {
+		signers, witnesses, err := reqParams[2].GetSignersWithWitnesses()
+		if err != nil {
+			return nil, response.ErrInvalidParams
+		}
+		tx.Signers = signers
+		tx.Scripts = witnesses
+	}
+	if len(tx.Signers) == 0 {
```
Comment (fyrchik) [inline pkg/rpc/server/server.go:1153]: C# module passes `null` transaction if signers are not present.

**Review by fyrchik — CHANGES_REQUESTED** [review]
We also need to change `Client.AddNetworkFee()` to use this RPC instead of `invokefunction`.

**Comment by roman-khimov** [comment]
@fyrchik, please review.

**Review by fyrchik — APPROVED** [review]

**Pull request merged**